server/libs/auth.js

```diff
@@ -0,0 +1,95 @@
+//
+// SecureImage
+//
+// Copyright © 2018 Province of British Columbia
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Created by Jason Leach on 2018-02-01.
+//
+
+/* eslint-env es6 */
+
+'use strict';
+
+import request from 'request';
+import jwt from 'jsonwebtoken';
+import pemFromModAndExponent from 'rsa-pem-from-mod-exp';
+
+const verifyToken = clientAccessToken => new Promise((resolve, reject) => {
+  request.get('https://dev-sso.pathfinder.gov.bc.ca/auth/realms/mobile/protocol/openid-connect/certs', {
+
+  }, (err, res, certsBody) => {
+    if (err) {
+      reject(err);
+      return;
+    }
+
+    const certsJson = JSON.parse(certsBody).keys[0];
+
+    const modulus = certsJson.n;
+    const exponent = certsJson.e;
+    const algorithm = certsJson.alg;
+
+    if (!modulus) {
+      reject(new Error('No modulus'));
+      return;
+    }
+
+    if (!exponent) {
+      reject(new Error('No exponent'));
+      return;
+    }
+
+    if (!algorithm) {
+      reject(new Error('No algorithm'));
+      return;
+    }
+
+    // build a certificate
+    const pem = pemFromModAndExponent(modulus, exponent);
+
+    // verify
+    jwt.verify(clientAccessToken, pem, { algorithms: [algorithm] }, (verifyErr, verifyResult) => {
+      if (verifyErr) {
+        reject(verifyErr);
+        return;
+      }
+      resolve(verifyResult);
+    });
+  });
+});
+
+function sendError(res, statusCode, message) {
```

Comment:
Lets put the non-exported functions all together at the top or bottom for rather than salt and peppering them in between the exports.

Reply:
So
-------------------
functions
-------------------
const promises
-------------------
exported promises
-------------------
?

Reply:
in this case I'd just stick with the fat-arrow function so cover it to `const sendError =  (res, statusCode, message) => {}` so that we're using a convention; if you need to use the function keyword the then I'd put them at the top. This is just a convention, no other impact other than it makes it easier for us to read.